opus/application/static_media/js/cart.js

```diff
@@ -112,6 +146,79 @@ var o_cart = {
         });
     },
 
+    updateDownloadFileInfo: function() {
+        /**
+         * Update file info after selecting/deselecting download options.
+         */
+        o_cart.showDownloadSizeSpinner();
+        let add_to_url = o_cart.getDownloadFiltersChecked();
+        o_cart.lastRequestNo++;
+        let url = "/opus/__cart/status.json?reqno=" + o_cart.lastRequestNo + "&" + add_to_url + "&download=1";
+        $.getJSON(url, function(info) {
+            if (info.reqno < o_cart.lastRequestNo) {
+                return;
+            }
+            o_cart.hideDownloadSizeSpinner(info.total_download_size_pretty);
+        });
+    },
+
+    updateCheckboxes: function(target, checkboxesOptions) {
+        /**
+         * Update checkboxes and disable/enable select all & deselect all
+         * buttons correspondingly.
+         */
+        if (($(target).hasClass("op-cart-select-btn") ||
+             $(target).hasClass("op-cart-select-all-btn"))) {
+            for (let checkbox of checkboxesOptions) {
+                $(checkbox).prop("checked", true);
+            }
+        } else if (($(target).hasClass("op-cart-deselect-btn") ||
+                    $(target).hasClass("op-cart-deselect-all-btn"))) {
+            for (let checkbox of checkboxesOptions) {
+                $(checkbox).prop("checked", false);
+            }
+        }
+        $(target).prop("disabled", true);
+        $(target).siblings().prop("disabled", false);
+        o_cart.updateDownloadFileInfo();
+    },
+
+    updateSelectDeselectBtn: function(isAllSelected, isAllDeselected, selectBtn, deselectBtn) {
+        /**
+         * Enable/disable select all and deselect all buttons based on
+         * parameters allSelected & allDeselected passed in.
```

Comment:
Actually the parameters are named isAllSelected etc. Also more clear to say "Enable/disable one pair of select all and deselect all buttons..."